### DISCUSSION TITLE
Importing triggers from user-generated file: Id issues ?

### DISCUSSION THREAD

**Discussion by NazzTazz** [bitfocus/companion#2870] [Q&A]
### Is this a feature relevant to companion itself, and not a module?

- [X] I believe this to be a feature for companion, not a module

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the feature

For a project I need to set 512 triggers in companion (wrapping midi messages to illuminate keys on a xkeys-128, 4 states, off blue red magenta, 4*128=512).

I tried to understand how to import triggers, simply reading an exported trigger file.

Triggers, Events, Action_sets have Ids like "I4uIFOG3AxS5k0Z5u5IZl", obviously base64, but I didn't succeed retreiving relevant data (tried UTF8, UTF7, and the other usual suspects).

Is there a way (API ?) to add triggers without using web interface ? Any clue on how Ids are crafted ? thanks

**Comment by dnmeid** [comment]
This is not an issue, but a question. I transfer it to discussions.

**Reply by NazzTazz** [reply]
I'm sorry, first post here, should have paid more attention !

**Comment by Julusian** [comment]
The Ids don't have any meaning, they are randomly generated strings.

There isn't an api to edit anything inside of companion. The best option would be to generate one of the exports, but that requires knowledge of how the files are structured which isnt documented. Or you could try to poke around the socketio to emulate what the ui does, but that undergoes breaking changes every now and then as it is an internal api designed solely based on what is needed for the ui

**Reply by NazzTazz** [reply]
Thanks dude, I figured out Ids don't mean anything. Currently dumping my whole companion config file from a LUA script, will share it here when it works !
